model_api/cpp/tilers/src/instance_segmentation.cpp

```diff
@@ -55,11 +55,13 @@ InstanceSegmentationTiler::InstanceSegmentationTiler(std::shared_ptr<ImageModel>
 
     postprocess_semantic_masks = get_from_any_maps("postprocess_semantic_masks",
                                                    configuration, extra_config, postprocess_semantic_masks);
+    max_pred_number = get_from_any_maps("max_pred_number", configuration, extra_config, max_pred_number);
 }
 
-std::unique_ptr<ResultBase> InstanceSegmentationTiler::run(const ImageInputData& inputData) {
+std::unique_ptr<InstanceSegmentationResult> InstanceSegmentationTiler::run(const ImageInputData& inputData) {
     auto setter = MaskRCNNModelParamsSetter(model);
-    return TilerBase::run(inputData);
+    auto result = this->run_impl(inputData);
```

Comment:
The issue is that we can't have several overloads of a method with similar args and covariant return types (if those are returned by value).
To overcome that, I've renamed the method in the base implementation and inherited IS tiler from the base one instead of DetectionTiler.
In async cpp PR I faced the same issue with batch inference method, so I had to introduce `inferBatchImage` method.